st2reactor/st2reactor/garbage_collector/base.py

```diff
@@ -196,7 +196,9 @@ def _purge_action_executions(self):
         timestamp_str = isotime.format(dt=timestamp)
         LOG.info('Deleting action executions older than: %s' % (timestamp_str))
 
-        assert timestamp < utc_now
+        if timestamp >= utc_now:
+            raise ValueError(f'Calculated timestamp ({timestamp}) is' %
+                            'later than now in UTC ({utc_now}).')
```

Comment:
Please do not combine f-strings with percent interpolation.

Same issue with the exception messages further down on lines 222/223 and 247/248.